app/web/src/newhotness/logic_composables/watched_form.ts

```diff
@@ -82,17 +82,23 @@ export const useWatchedForm = <Data>(label: string) => {
     });
     const wForm = useForm({
       ...opts,
-      onSubmit: (props) => {
+      onSubmit: async (props) => {
         span = tracer.startSpan("watchedForm");
         span.setAttributes({
           workspaceId: ctx.workspacePk.value,
           changeSetId: ctx.changeSetId.value,
           form: label,
         });
+
+        // Submit the form and start waiting for the data to come back
         start = Date.now();
-        onSubmit(props);
+        await onSubmit(props);
         bifrosting.value = true;
         rainbow.add(ctx.changeSetId.value, label);
```

Comment:
now that we're awaiting, we should move bifrosting and rainbow add _before_ the `onSubmit`